src/tools/fuzzing.h

```diff
@@ -2636,6 +2667,154 @@ class TranslateToFuzzReader {
+    auto* ret = builder.makeTry(makeMaybeBlock(type), makeCatch(type), type);
+    hangStack.pop_back();
+    return ret;
+  }
+
+  Expression* makeCatch(Type type) {
+    // exnref.pop is a pseudo instruction that should follow 'catch'
+    Expression* pop = builder.makePop(Type::exnref);
+
+    // With a small probability, just rethrow it and get done
+    if (oneIn(10)) {
+      return builder.makeRethrow(pop);
+    }
+
+    // Make a block. Note that we don't insert this break in breakableStack:
+    // Because 'exnref.pop' should follow 'catch' and we don't support blocks
+    // taking values yet, if there are instructions that break to this block,
+    // this block cannot be removed when writing binary, which will be a
+    // problem. We also don't make a label for this block.
+    auto* ret = builder.makeBlock();
+    ret->type = type;
+
+    // Now that we have to consume exnref.pop. We try one of these four things:
+    // 1. Drop it
+    // 2. Set to a local
+    // 3. Set to a global
+    // 4. Use it as an argument to a function
+    Expression* firstExpr = nullptr;
+    switch (upTo(4)) {
+      case 0:
+        firstExpr = builder.makeDrop(pop);
+        break;
+      case 1: {
+        auto& locals = typeLocals[Type::exnref];
+        if (locals.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          firstExpr = builder.makeLocalSet(pick(locals), pop);
+        }
+        break;
+      }
+      case 2: {
+        auto& globals = globalsByType[Type::exnref];
+        if (globals.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          firstExpr = builder.makeGlobalSet(pick(globals), pop);
+        }
+        break;
+      }
+      case 3: {
+        if (wasm.functions.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          int tries = TRIES;
+          Function* target = nullptr;
+          while (tries-- > 0) {
+            target = pick(wasm.functions).get();
+            if (target->sig.results != Type::none ||
+                target->sig.params.size() == 0 ||
+                *target->sig.params.expand().begin() != Type::exnref) {
+              continue;
+            }
+            // we found a function that satisfies all conditions!
+            std::vector<Expression*> args;
+            args.push_back(pop);
+            for (unsigned i = 1; i < target->sig.params.size(); i++) {
+              args.push_back(make(target->sig.params.expand()[i]));
+            }
+            firstExpr = builder.makeCall(target->name, args, type, false);
+          }
+          // we failed to find something
+          firstExpr = builder.makeDrop(pop);
+        }
+        break;
+      }
+    }
+
+    // Populate the rest of the block.
+    ret->list.push_back(firstExpr);
+    Index num = getNumBlockInstructions();
+    while (num > 0 && !finishedInput) {
+      ret->list.push_back(make(Type::none));
+      num--;
+    }
```

Comment:
Would it be simpler to just `makeBlock` then prepend the pop-handling instruction to the list of children?

Reply:
`makeBlock` does things we shouldn't do here, such as, it [makes a label](https://github.com/WebAssembly/binaryen/blob/dfe473e6af0a31cad7a7b26f5dead358d9bbf536/src/tools/fuzzing.h#L976), [adds itself to a breakable stack](https://github.com/WebAssembly/binaryen/blob/dfe473e6af0a31cad7a7b26f5dead358d9bbf536/src/tools/fuzzing.h#L977), and sometimes [it ends the block with an unreachable break](https://github.com/WebAssembly/binaryen/blob/dfe473e6af0a31cad7a7b26f5dead358d9bbf536/src/tools/fuzzing.h#L995-L1001). (For the unreachable break thing, we do a similar thing with a `rethrow`.)